src/details/integral.cpp

```diff
@@ -0,0 +1,26 @@
+#include <path_tracking_pid/details/integral.hpp>
+
+namespace path_tracking_pid::details
+{
+Integral::Integral(double windup_limit) : windup_limit_(windup_limit) {}
+
+void Integral::configure(double windup_limit) { windup_limit_ = windup_limit; }
+
+double Integral::filter(double u, double step_size)
+{
+  // save history
+  u_.push(u);
+  y_.push(u);  // increase index so the math below looks correct
+
+  auto T = step_size;
+  y_[0] = T / 2 * (u_[0] + u_[1]) + y_[1];
```

Comment:
This is different math than before, right? Previously, the integral step just added the `new input * T` to the previous output (backward Euler method). Now, you're adding the `average of current and previous inputs * T` to the previous output (trapezoidal method).
 1. Why this change?
 2. Maybe it's good to document the integral method you're using in the header?

Reply:
I've did the same discretezation method (tustin) as the lowpass to prevent confusion. I should definitely add a comment about that.

Reply:
I've added this comment:
> // A continous time integrator was discretized with Tustin's method. For a mathematical
  // explanation, see doc/integral_tustin.ipynb